draft-ietf-rats-eat.md

```diff
@@ -1198,103 +1150,42 @@ The string identifying the JSON-encoded token MUST be one of the following:
+#### Submodule Claims-Set
 
-The integrity protection for the larger claims sets will not be as secure as those originating in hardware block, but the key material and hardware-based claims will be.
-It is possible for the hardware to enforce hardware access control (memory protection)  on the digest registers so that some of the larger claims can be more secure.
-For example, one register may be writable only by the TEE, so the detached claims from the TEE will have TEE-level security.
+The Claims-Set type provides a means of representing claims from a submodule that does not have its own attesting environment,
+i.e., it has no keys distinct from the attester producing the surrounding token. Claims are represented as a Claims-Set. Submodule claims represented in this way are secured by the same
+mechanism as the enclosing token (e.g., it is signed by the same attestation key).
 
-The data type for this type of submodule MUST be an array
-It contains two data items, a hash algorithm identifier and a byte string containing the digest.
+The encoding of a submodule Claims-Set MUST be the same as the encoding as the surrounding EAT, e.g., all submodule Claims-Sets in a CBOR-encoded token must be CBOR-encoded.
 
-The hash algorithm identifier is always from the COSE Algorithm registry, {{IANA.COSE.Algorithms}}.
-Either the integer or string identifier may be used.
-The hash algorithm identifier is never from the JOSE Algorithm registry.
+#### Detached Submodule Digest
 
-When decoding a CBOR format token, the detached digest type is distinguished from the other types by it being an array.
-In CBOR encoded tokens none of other submodule types are arrays.
-
-When decoding a JSON format token, a little more work is required because both the nested token and detached digest types are an array.
-To distinguish the nested token from the detached digest, the first element in the array is examined.
-If it is "JWT" or "BUNDLE", then the submodule is a nested token.
-Otherwise it will contain an algorithm identifier and is a detached digest.
-
-A detached EAT bundle, described in {{DEB}}, may be used to convey detached claims sets and the token with their detached digests.
-EAT, however, doesn't require use of a detached EAT bundle.
-Any other protocols may be used to convey detached claims sets and the token with their detached digests.
-Note that since detached Claims-Sets are signed, protocols conveying them must make sure they are not modified in transit.
-
-~~~~CDDL
-{::include nc-cddl/detached-digest.cddl}
-~~~~
+The Detached-Submodule-Digest type is similar to a submodule Claims-Set, except a digest of the Claims-Set is included in the claim with the Claims-Set contents conveyed separately. The separately conveyed Claims-Set is called a detached claims set. The input to the digest algorithm is directly the CBOR or JSON-form of the Claims-Set for the submodule. There is no byte-string wrapping or base 64 encoding.
```

Comment:
I kind think it should be "separately-conveyed" with the hyphen, but my grammar checker disagrees.  Either way is fine for me.

Reply:
Changed